client.go

```diff
@@ -132,8 +134,23 @@ func (c *GraphQLClient) Request(ctx context.Context, url string, request *Reques
 	}
 
 	var buf bytes.Buffer
-	if err := json.NewEncoder(&buf).Encode(request); err != nil {
-		return traceErr(fmt.Errorf("unable to encode request body: %w", err))
+	contentType := "application/json; charset=utf-8"
+	requestContentType := ""
+	if request.Headers != nil {
+		requestContentType = request.Headers.Get("Content-Type")
+	}
+	if !strings.Contains(requestContentType, "multipart") {
```

Comment:
```suggestion
	if requestContentType == "multipart/form-data" {
```
How about inverting this if statement, so it treats it as multipart if it's certain it can, and have everything else as the default case.

I don't think you need to check for other `multipart` mime types as the spec only specifies behaviour for `multipart/form-data`

Reply:
@pkqk sure, I'll inverse the statement. However, we should probably still use `contains` as `Content-Type` header will also have boundary in case of multipart:
```
"request.content-type":"multipart/form-data; boundary=--------------------------208172004141002940919311"
```

Reply:
I've inverted the statement, changed `Contains` to `HasPrefix` and specified `multipart/form-data`.